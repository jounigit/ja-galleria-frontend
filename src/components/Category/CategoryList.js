import React, { useState, useEffect } from 'react'
import axios from 'axios'

import ListItem from './ListItem'
const url = 'http://localhost:8000/api/categories'

const CategoryList = () => {
  const [data, setData] = useState([])
  const [isLoading, setIsLoading] = useState(false)
  const [isError, setIsError] = useState(false)

  useEffect(() => {
    const FetchData = async () => {
      setIsError(false)
      setIsLoading(true)

      try {
        const result = await axios(url)
        setData(result.data.data)
      } catch (error) {
        setIsError(true)
      }
      setIsLoading(false)
    }
    FetchData()
  }, [])

  console.log('DATA -- ', data)

  return (
    <div>
      <h2>Kategoriat</h2>
      {isError && <div>Something went wrong ...</div>}
      {isLoading ? (
        <div>Loading...</div>
      ) : (
        <ul>
          { data.map(category =>
            <ListItem key={category.id} category={category} />
          ) }
        </ul>
      )}
    </div>
  )
}

export default CategoryList