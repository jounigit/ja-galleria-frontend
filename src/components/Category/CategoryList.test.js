import React from 'react'
// import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import { render } from '@testing-library/react'
import CategoryList from './CategoryList'
import { BrowserRouter } from 'react-router-dom'

let container

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})


test('should render all categories ', async () => {
  await act(async () => {
    render(
      <BrowserRouter>
        <CategoryList />
      </BrowserRouter>, container)
  })

  container.debug()

  expect(container).toHaveTextContent('loading..')
  await act(async () => {
    render(
      <BrowserRouter>
        <CategoryList />
      </BrowserRouter>, container)
  })

  // const categories = container.querySelectorAll('h3')
  // expect(categories.length).toBe(3)

  // expect(container).toHaveTextContent('Categoria 1')

  // expect(container).toHaveTextContent('Eaque a culpa et qui nam fugiat.')
})


