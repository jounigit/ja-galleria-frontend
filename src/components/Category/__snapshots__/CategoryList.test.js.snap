// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all categories  1`] = `
<div>
  Loading...
</div>
`;
